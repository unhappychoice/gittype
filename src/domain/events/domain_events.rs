use super::Event;
use std::any::Any;

// Unified domain event enum for pattern matching
#[derive(Debug, Clone)]
pub enum DomainEvent {
    KeyPressed { key: char, position: usize },
    StageStarted { start_time: std::time::Instant },
    StagePaused,
    StageResumed,
    StageFinalized,
    StageSkipped,
    ChallengeLoaded { text: String, source_path: String },
}

impl Event for DomainEvent {
    fn as_any(&self) -> &dyn Any {
        self
    }
}
