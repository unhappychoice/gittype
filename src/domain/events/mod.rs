use std::any::{Any, TypeId};
use std::collections::HashMap;
use std::sync::{Arc, RwLock};

pub mod domain_events;

pub trait Event: Send + Sync + 'static {
    fn as_any(&self) -> &dyn Any;
}

pub trait EventHandler<E: Event>: Send + Sync {
    fn handle(&self, event: &E);
}

type BoxedHandler = Arc<dyn Fn(&dyn Event) + Send + Sync>;

pub struct EventBus {
    subscribers: Arc<RwLock<HashMap<TypeId, Vec<BoxedHandler>>>>,
}

impl EventBus {
    pub fn new() -> Self {
        Self {
            subscribers: Arc::new(RwLock::new(HashMap::new())),
        }
    }

    pub fn publish<E: Event>(&self, event: E) {
        let type_id = TypeId::of::<E>();

        // Clone Arc<handlers> while holding the lock, then release it before calling them
        let handlers: Vec<BoxedHandler> = {
            let subscribers = self.subscribers.read().unwrap();
            let h = subscribers.get(&type_id)
                .map(|h| h.clone())
                .unwrap_or_default();
            h
        }; // Lock is released here

        // Call handlers without holding the lock
        for handler in handlers.iter() {
            handler(&event);
        }
    }

    pub fn subscribe<E: Event, F>(&self, handler: F)
    where
        F: Fn(&E) + Send + Sync + 'static,
    {
        let type_id = TypeId::of::<E>();
        let mut subscribers = self.subscribers.write().unwrap();

        let boxed_handler: BoxedHandler = Arc::new(move |event: &dyn Event| {
            if let Some(concrete_event) = event.as_any().downcast_ref::<E>() {
                handler(concrete_event);
            }
        });

        subscribers
            .entry(type_id)
            .or_insert_with(Vec::new)
            .push(boxed_handler);
    }
}

impl Default for EventBus {
    fn default() -> Self {
        Self::new()
    }
}

impl Clone for EventBus {
    fn clone(&self) -> Self {
        Self {
            subscribers: Arc::clone(&self.subscribers),
        }
    }
}
