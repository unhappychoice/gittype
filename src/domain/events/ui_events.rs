use super::Event;
use std::any::Any;

#[derive(Debug, Clone)]
pub struct RenderRequested {
    pub screen_name: String,
}

impl Event for RenderRequested {
    fn as_any(&self) -> &dyn Any {
        self
    }
}

#[derive(Debug, Clone)]
pub struct ScreenChanged {
    pub from: String,
    pub to: String,
}

impl Event for ScreenChanged {
    fn as_any(&self) -> &dyn Any {
        self
    }
}

#[derive(Debug, Clone)]
pub struct UserInputReceived {
    pub input: String,
}

impl Event for UserInputReceived {
    fn as_any(&self) -> &dyn Any {
        self
    }
}

#[derive(Debug, Clone)]
pub struct ConfigUpdated {
    pub key: String,
    pub value: String,
}

impl Event for ConfigUpdated {
    fn as_any(&self) -> &dyn Any {
        self
    }
}
