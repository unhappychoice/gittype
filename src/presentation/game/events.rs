use std::any::Any;

// Re-export ScreenTransition as NavigateTo event
pub use super::models::screen::ScreenTransition as NavigateTo;

/// Event emitted when user requests to exit the application (Ctrl+C)
#[derive(Clone, Debug)]
pub struct ExitRequested;

impl crate::domain::events::Event for ExitRequested {
    fn as_any(&self) -> &dyn Any {
        self
    }
}
