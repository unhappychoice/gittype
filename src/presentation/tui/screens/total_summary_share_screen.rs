use crate::domain::events::EventBusInterface;
use crate::domain::models::TotalResult;
use crate::domain::services::scoring::{TotalCalculator, TotalTracker, GLOBAL_TOTAL_TRACKER};
use crate::infrastructure::browser;
use crate::presentation::game::events::NavigateTo;
use crate::presentation::sharing::SharingPlatform;
use crate::presentation::tui::views::SharingView;
use crate::presentation::tui::{Screen, ScreenDataProvider, ScreenType, UpdateStrategy};
use crate::{GitTypeError, Result};
use crossterm::event::{self};
use ratatui::Frame;
use std::sync::RwLock;
use std::sync::{Arc, Mutex};

pub struct TotalSummaryShareData {
    pub total_result: TotalResult,
}

pub struct TotalSummaryShareDataProvider {
    total_tracker: Arc<Mutex<Option<TotalTracker>>>,
}

impl ScreenDataProvider for TotalSummaryShareDataProvider {
    fn provide(&self) -> Result<Box<dyn std::any::Any>> {
        let total_result = self
            .total_tracker
            .lock()
            .map_err(|_| GitTypeError::TerminalError("Failed to lock TotalTracker".to_string()))?
            .as_ref()
            .ok_or_else(|| GitTypeError::TerminalError("No total tracker available".to_string()))
            .map(TotalCalculator::calculate)?;

        Ok(Box::new(TotalSummaryShareData { total_result }))
    }
}

#[derive(Debug)]
pub enum ShareAction {
    Back,
    Exit,
}

pub trait TotalSummaryShareScreenInterface: Screen {}

#[derive(shaku::Component)]
#[shaku(interface = TotalSummaryShareScreenInterface)]
pub struct TotalSummaryShareScreen {
    #[shaku(default)]
    total_result: RwLock<TotalResult>,
    #[shaku(default)]
    fallback_url: RwLock<Option<(String, SharingPlatform)>>,
    #[shaku(default)]
    last_fallback_state: RwLock<bool>,
    #[shaku(inject)]
    event_bus: Arc<dyn EventBusInterface>,
}

impl TotalSummaryShareScreen {
    pub fn new(event_bus: Arc<dyn EventBusInterface>) -> Self {
        Self {
            total_result: RwLock::new(TotalResult::new()),
            fallback_url: RwLock::new(None),
            last_fallback_state: RwLock::new(false),
            event_bus,
        }
    }

    pub fn set_total_result(&self, total_result: TotalResult) {
        *self.total_result.write().unwrap() = total_result;
    }

    fn handle_share_platform(&self, platform: SharingPlatform) -> Result<()> {
        let total_result = self.total_result.read().unwrap();
        let text = total_result.create_share_text();
        let url = self.generate_share_url(&text, &platform);

        match self.open_browser(&url) {
            Ok(()) => {
                self.event_bus.as_event_bus().publish(NavigateTo::Pop);
                Ok(())
            }
            Err(_) => {
                log::warn!(
                    "Failed to open browser for {}. URL: {}",
                    platform.name(),
                    url
                );
                *self.fallback_url.write().unwrap() = Some((url.clone(), platform));
                Ok(())
            }
        }
    }

    fn generate_share_url(&self, text: &str, platform: &SharingPlatform) -> String {
        match platform {
            SharingPlatform::X => {
                format!(
                    "https://x.com/intent/tweet?text={}",
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::Reddit => {
                let total_result = self.total_result.read().unwrap();
                let title = format!(
                    "Just demolished {} keystrokes total in gittype! Score: {:.0}, CPM: {:.0}",
                    total_result.total_keystrokes,
                    total_result.total_score,
                    total_result.overall_cpm
                );
                format!(
                    "https://www.reddit.com/submit?title={}&selftext=true&text={}",
                    urlencoding::encode(&title),
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::LinkedIn => {
                format!(
                    "https://www.linkedin.com/feed/?shareActive=true&mini=true&text={}",
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::Facebook => {
                format!(
                    "https://www.facebook.com/sharer/sharer.php?u={}&quote={}",
                    urlencoding::encode("https://github.com/unhappychoice/gittype"),
                    urlencoding::encode(text)
                )
            }
        }
    }

    fn open_browser(&self, url: &str) -> crate::Result<()> {
        browser::open_url(url)
            .map_err(|e| GitTypeError::TerminalError(format!("Failed to open browser: {}", e)))
    }
}

pub struct TotalSummaryShareScreenProvider;

impl shaku::Provider<crate::presentation::di::AppModule> for TotalSummaryShareScreenProvider {
    type Interface = TotalSummaryShareScreen;

    fn provide(
        module: &crate::presentation::di::AppModule,
    ) -> std::result::Result<Box<Self::Interface>, Box<dyn std::error::Error>> {
        use shaku::HasComponent;
        let event_bus: std::sync::Arc<dyn crate::domain::events::EventBusInterface> =
            module.resolve();
        Ok(Box::new(TotalSummaryShareScreen::new(event_bus)))
    }
}

impl Screen for TotalSummaryShareScreen {
    fn get_type(&self) -> ScreenType {
        ScreenType::TotalSummaryShare
    }

    fn default_provider() -> Box<dyn crate::presentation::tui::ScreenDataProvider>
    where
        Self: Sized,
    {
        Box::new(TotalSummaryShareDataProvider {
            total_tracker: GLOBAL_TOTAL_TRACKER.clone(),
        })
    }

    fn init_with_data(&self, data: Box<dyn std::any::Any>) -> Result<()> {
        *self.last_fallback_state.write().unwrap() = false;

        let data = data.downcast::<TotalSummaryShareData>()?;
        *self.total_result.write().unwrap() = data.total_result;

        Ok(())
    }

    fn handle_key_event(&self, key_event: event::KeyEvent) -> Result<()> {
        use crossterm::event::{KeyCode, KeyModifiers};
        match key_event.code {
            KeyCode::Char('1') => self.handle_share_platform(SharingPlatform::X),
            KeyCode::Char('2') => self.handle_share_platform(SharingPlatform::Reddit),
            KeyCode::Char('3') => self.handle_share_platform(SharingPlatform::LinkedIn),
            KeyCode::Char('4') => self.handle_share_platform(SharingPlatform::Facebook),
            KeyCode::Esc => {
                let fallback_url = self.fallback_url.read().unwrap();
                if fallback_url.is_some() {
                    drop(fallback_url);
                    *self.fallback_url.write().unwrap() = None;
                    Ok(())
                } else {
                    self.event_bus.as_event_bus().publish(NavigateTo::Pop);
                    Ok(())
                }
            }
            KeyCode::Char('c') if key_event.modifiers.contains(KeyModifiers::CONTROL) => {
                self.event_bus.as_event_bus().publish(NavigateTo::Exit);
                Ok(())
            }
            _ => Ok(()),
        }
    }

    fn render_ratatui(&self, frame: &mut Frame) -> Result<()> {
        let fallback_url = self.fallback_url.read().unwrap();
        if let Some((url, platform)) = &*fallback_url {
            SharingView::render_fallback_url(frame, url, platform);
        } else {
            let total_result = self.total_result.read().unwrap();
            SharingView::render_menu(frame, &total_result);
        }
        Ok(())
    }

    fn get_update_strategy(&self) -> UpdateStrategy {
        UpdateStrategy::TimeBased(std::time::Duration::from_millis(200))
    }

    fn update(&self) -> Result<bool> {
        Ok(true)
    }

    fn as_any(&self) -> &dyn std::any::Any {
        self
    }
}

impl TotalSummaryShareScreenInterface for TotalSummaryShareScreen {}
