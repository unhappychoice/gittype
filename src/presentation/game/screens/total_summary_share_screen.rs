use crate::presentation::game::models::{Screen, ScreenTransition, UpdateStrategy};
use crate::presentation::game::views::SharingView;
use crate::domain::models::TotalResult;
use crate::presentation::sharing::SharingPlatform;
use crate::Result;
use crossterm::{
    event::{self},
    execute,
    terminal::{self, ClearType},
};
use std::io::Stdout;

#[derive(Debug)]
pub enum ShareAction {
    Back,
    Exit,
}

pub struct TotalSummaryShareScreen {
    total_result: TotalResult,
    fallback_url: Option<(String, SharingPlatform)>,
    last_fallback_state: bool,
}

impl TotalSummaryShareScreen {
    pub fn new(total_result: TotalResult) -> Self {
        Self {
            total_result,
            fallback_url: None,
            last_fallback_state: false,
        }
    }

    fn handle_share_platform(&mut self, platform: SharingPlatform) -> Result<ScreenTransition> {
        let text = self.total_result.create_share_text();
        let url = self.generate_share_url(&text, &platform);

        match self.open_browser(&url) {
            Ok(()) => Ok(ScreenTransition::Pop),
            Err(_) => {
                log::warn!(
                    "Failed to open browser for {}. URL: {}",
                    platform.name(),
                    url
                );
                self.fallback_url = Some((url.clone(), platform));
                Ok(ScreenTransition::None)
            }
        }
    }

    fn generate_share_url(&self, text: &str, platform: &SharingPlatform) -> String {
        match platform {
            SharingPlatform::X => {
                format!(
                    "https://x.com/intent/tweet?text={}",
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::Reddit => {
                let title = format!(
                    "Just demolished {} keystrokes total in gittype! Score: {:.0}, CPM: {:.0}",
                    self.total_result.total_keystrokes,
                    self.total_result.total_score,
                    self.total_result.overall_cpm
                );
                format!(
                    "https://www.reddit.com/submit?title={}&selftext=true&text={}",
                    urlencoding::encode(&title),
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::LinkedIn => {
                format!(
                    "https://www.linkedin.com/feed/?shareActive=true&mini=true&text={}",
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::Facebook => {
                format!(
                    "https://www.facebook.com/sharer/sharer.php?u={}&quote={}",
                    urlencoding::encode("https://github.com/unhappychoice/gittype"),
                    urlencoding::encode(text)
                )
            }
        }
    }

    fn open_browser(&self, url: &str) -> crate::Result<()> {
        open::that(url).map_err(|e| {
            crate::domain::error::GitTypeError::TerminalError(format!("Failed to open browser: {}", e))
        })
    }
}

impl Screen for TotalSummaryShareScreen {
    fn init(&mut self) -> crate::Result<()> {
        self.last_fallback_state = false;
        Ok(())
    }

    fn handle_key_event(&mut self, key_event: event::KeyEvent) -> Result<ScreenTransition> {
        use crossterm::event::{KeyCode, KeyModifiers};
        match key_event.code {
            KeyCode::Char('1') => self.handle_share_platform(SharingPlatform::X),
            KeyCode::Char('2') => self.handle_share_platform(SharingPlatform::Reddit),
            KeyCode::Char('3') => self.handle_share_platform(SharingPlatform::LinkedIn),
            KeyCode::Char('4') => self.handle_share_platform(SharingPlatform::Facebook),
            KeyCode::Esc => {
                if self.fallback_url.is_some() {
                    self.fallback_url = None;
                    Ok(ScreenTransition::None)
                } else {
                    Ok(ScreenTransition::Pop)
                }
            }
            KeyCode::Char('c') if key_event.modifiers.contains(KeyModifiers::CONTROL) => {
                Ok(ScreenTransition::Exit)
            }
            _ => Ok(ScreenTransition::None),
        }
    }

    fn render_crossterm_with_data(
        &mut self,
        stdout: &mut Stdout,
        _session_result: Option<&crate::domain::models::SessionResult>,
        _total_result: Option<&TotalResult>,
    ) -> Result<()> {
        let current_fallback_state = self.fallback_url.is_some();

        if current_fallback_state != self.last_fallback_state {
            execute!(stdout, terminal::Clear(ClearType::All))?;
            self.last_fallback_state = current_fallback_state;
        }

        let (terminal_width, terminal_height) = terminal::size()?;
        let center_row = terminal_height / 2;
        let center_col = terminal_width / 2;

        if let Some((url, platform)) = &self.fallback_url {
            SharingView::render_fallback_url(stdout, url, platform, center_col, center_row)?;
        } else {
            SharingView::render_menu(stdout, &self.total_result, center_col, center_row)?;
        }
        Ok(())
    }

    fn get_update_strategy(&self) -> UpdateStrategy {
        UpdateStrategy::TimeBased(std::time::Duration::from_millis(200))
    }

    fn update(&mut self) -> Result<bool> {
        Ok(true)
    }

    fn as_any(&self) -> &dyn std::any::Any {
        self
    }

    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
        self
    }
}
