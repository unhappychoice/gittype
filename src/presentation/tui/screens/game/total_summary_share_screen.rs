use crate::domain::events::EventBus;
use crate::domain::models::TotalResult;
use crate::domain::services::scoring::{TotalCalculator, TotalTracker, GLOBAL_TOTAL_TRACKER};
use crate::infrastructure::browser;
use crate::presentation::game::events::NavigateTo;
use crate::presentation::game::views::SharingView;
use crate::presentation::game::{Screen, ScreenDataProvider, ScreenType, UpdateStrategy};
use crate::presentation::sharing::SharingPlatform;
use crate::{GitTypeError, Result};
use crossterm::event::{self};
use ratatui::Frame;
use std::sync::{Arc, Mutex};

pub struct TotalSummaryShareData {
    pub total_result: TotalResult,
}

pub struct TotalSummaryShareDataProvider {
    total_tracker: Arc<Mutex<Option<TotalTracker>>>,
}

impl ScreenDataProvider for TotalSummaryShareDataProvider {
    fn provide(&self) -> Result<Box<dyn std::any::Any>> {
        let total_result = self
            .total_tracker
            .lock()
            .map_err(|_| GitTypeError::TerminalError("Failed to lock TotalTracker".to_string()))?
            .as_ref()
            .ok_or_else(|| GitTypeError::TerminalError("No total tracker available".to_string()))
            .map(TotalCalculator::calculate)?;

        Ok(Box::new(TotalSummaryShareData { total_result }))
    }
}

#[derive(Debug)]
pub enum ShareAction {
    Back,
    Exit,
}

pub struct TotalSummaryShareScreen {
    total_result: TotalResult,
    fallback_url: Option<(String, SharingPlatform)>,
    last_fallback_state: bool,
    event_bus: EventBus,
}

impl TotalSummaryShareScreen {
    pub fn new(event_bus: EventBus) -> Self {
        Self {
            total_result: TotalResult::new(),
            fallback_url: None,
            last_fallback_state: false,
            event_bus,
        }
    }

    pub fn set_total_result(&mut self, total_result: TotalResult) {
        self.total_result = total_result;
    }

    fn handle_share_platform(&mut self, platform: SharingPlatform) -> Result<()> {
        let text = self.total_result.create_share_text();
        let url = self.generate_share_url(&text, &platform);

        match self.open_browser(&url) {
            Ok(()) => {
                self.event_bus.publish(NavigateTo::Pop);
                Ok(())
            }
            Err(_) => {
                log::warn!(
                    "Failed to open browser for {}. URL: {}",
                    platform.name(),
                    url
                );
                self.fallback_url = Some((url.clone(), platform));
                Ok(())
            }
        }
    }

    fn generate_share_url(&self, text: &str, platform: &SharingPlatform) -> String {
        match platform {
            SharingPlatform::X => {
                format!(
                    "https://x.com/intent/tweet?text={}",
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::Reddit => {
                let title = format!(
                    "Just demolished {} keystrokes total in gittype! Score: {:.0}, CPM: {:.0}",
                    self.total_result.total_keystrokes,
                    self.total_result.total_score,
                    self.total_result.overall_cpm
                );
                format!(
                    "https://www.reddit.com/submit?title={}&selftext=true&text={}",
                    urlencoding::encode(&title),
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::LinkedIn => {
                format!(
                    "https://www.linkedin.com/feed/?shareActive=true&mini=true&text={}",
                    urlencoding::encode(text)
                )
            }
            SharingPlatform::Facebook => {
                format!(
                    "https://www.facebook.com/sharer/sharer.php?u={}&quote={}",
                    urlencoding::encode("https://github.com/unhappychoice/gittype"),
                    urlencoding::encode(text)
                )
            }
        }
    }

    fn open_browser(&self, url: &str) -> crate::Result<()> {
        browser::open_url(url)
            .map_err(|e| GitTypeError::TerminalError(format!("Failed to open browser: {}", e)))
    }
}

impl Screen for TotalSummaryShareScreen {
    fn get_type(&self) -> ScreenType {
        ScreenType::TotalSummaryShare
    }

    fn default_provider() -> Box<dyn crate::presentation::game::ScreenDataProvider>
    where
        Self: Sized,
    {
        Box::new(TotalSummaryShareDataProvider {
            total_tracker: GLOBAL_TOTAL_TRACKER.clone(),
        })
    }

    fn init_with_data(&mut self, data: Box<dyn std::any::Any>) -> Result<()> {
        self.last_fallback_state = false;

        let data = data.downcast::<TotalSummaryShareData>()?;
        self.total_result = data.total_result;

        Ok(())
    }

    fn handle_key_event(&mut self, key_event: event::KeyEvent) -> Result<()> {
        use crossterm::event::{KeyCode, KeyModifiers};
        match key_event.code {
            KeyCode::Char('1') => self.handle_share_platform(SharingPlatform::X),
            KeyCode::Char('2') => self.handle_share_platform(SharingPlatform::Reddit),
            KeyCode::Char('3') => self.handle_share_platform(SharingPlatform::LinkedIn),
            KeyCode::Char('4') => self.handle_share_platform(SharingPlatform::Facebook),
            KeyCode::Esc => {
                if self.fallback_url.is_some() {
                    self.fallback_url = None;
                    Ok(())
                } else {
                    self.event_bus.publish(NavigateTo::Pop);
                    Ok(())
                }
            }
            KeyCode::Char('c') if key_event.modifiers.contains(KeyModifiers::CONTROL) => {
                self.event_bus.publish(NavigateTo::Exit);
                Ok(())
            }
            _ => Ok(()),
        }
    }

    fn render_ratatui(&mut self, frame: &mut Frame) -> Result<()> {
        if let Some((url, platform)) = &self.fallback_url {
            SharingView::render_fallback_url(frame, url, platform);
        } else {
            SharingView::render_menu(frame, &self.total_result);
        }
        Ok(())
    }

    fn get_update_strategy(&self) -> UpdateStrategy {
        UpdateStrategy::TimeBased(std::time::Duration::from_millis(200))
    }

    fn update(&mut self) -> Result<bool> {
        Ok(true)
    }

    fn as_any(&self) -> &dyn std::any::Any {
        self
    }

    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
        self
    }
}
