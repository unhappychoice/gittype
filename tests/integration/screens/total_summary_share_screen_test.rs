use crate::integration::screens::mocks::total_summary_share_screen_mock::MockTotalSummaryShareDataProvider;
use gittype::domain::events::EventBus;
use gittype::presentation::game::screens::total_summary_share_screen::TotalSummaryShareScreen;

screen_snapshot_test!(
    test_total_summary_share_screen_snapshot,
    TotalSummaryShareScreen,
    TotalSummaryShareScreen::new(EventBus::new()),
    provider = MockTotalSummaryShareDataProvider
);
