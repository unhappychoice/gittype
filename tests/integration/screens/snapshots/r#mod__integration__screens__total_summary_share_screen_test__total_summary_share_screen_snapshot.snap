---
source: tests/integration/screens/total_summary_share_screen_test.rs
expression: output
---
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                                                                                        
                                                Share Your Total Results                                                
                                                                                                                        
                                                                                                                        
                        Score: 5000, CPM: 320, Keystrokes: 15000, Sessions: 25/30, Time: 60.0min                        
                                                                                                                        
                                                                                                                        
                                                          [1] X                                                         
                                                       [2] Reddit                                                       
                                                      [3] LinkedIn                                                      
                                                      [4] Facebook                                                      
                                                                                                                        
                                                                                                                        
                                                [ESC] Back to Exit Screen
